pkg/webhook/mutation/pod_mutator/pod_mutator.go

```diff
@@ -196,17 +201,20 @@ func setDynatraceInjectedAnnotation(mutationRequest *dtwebhook.MutationRequest)
 
 // createResponseForPod tries to format pod as json
 func createResponseForPod(ctx context.Context, pod *corev1.Pod, req admission.Request) admission.Response {
-	_, span := dtotel.StartSpan(ctx, otelName, "createResponseForPod")
+	ctx, span := dtotel.StartSpan(ctx, webhookotel.Tracer(), spanOptions()...)
 	defer span.End()
 
 	marshaledPod, err := json.MarshalIndent(pod, "", "  ")
 	if err != nil {
-		return silentErrorResponse(pod, err)
+		return silentErrorResponse(ctx, pod, err)
 	}
 	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
 }
 
-func silentErrorResponse(pod *corev1.Pod, err error) admission.Response {
+func silentErrorResponse(ctx context.Context, pod *corev1.Pod, err error) admission.Response {
+	_, span := dtotel.StartSpan(ctx, webhookotel.Tracer(), spanOptions()...)
+	defer span.End()
+
```

Comment:
why open a span for a function that does not really use the context?

I don't want to comment for every function where it occurs but my thoughts/questions are:
- why open spans for functions that are not even using the context (because they don't do anything that would really need it)?
- aren't we supposed to "record" errors to the span? which would be made hell if every function creates its on span, no ? to which span should the error belong to?

Reply:
If a function uses the context or not has no relevance for the question if it needs a span. The context merely carries the parent span. The new span is written to the context that is returned by `StartSpan`, and in this case it's never used as a parent span for other functions or to record errors. But if this function would have some potentially expensive calculations it would still make sense to have a span to measure the time it needs.
Now, with that being said, it's for sure debatable  if this function worth a span, but we also discussed to go overboard with spans in some code parts just to learn how this feels and looks like.

Reply:
And IMHO the error should always be recorded to a span that's opened in the function where it first appears. That way we have the best mapping to the error location. I'm not sure how this would look like, if all errors are recorded to say one global span. That's also something we have to learn.